pkg/autoscaler/metrics/stats_scraper.go

```diff
@@ -118,11 +133,29 @@ func newServiceScraperWithClient(
 	if revName == "" {
 		return nil, fmt.Errorf("no Revision label found for Metric %s", metric.Name)
 	}
+	svcName := metric.Labels[serving.ServiceLabelKey]
+	cfgName := metric.Labels[serving.ConfigurationLabelKey]
+
+	if err := view.Register(
```

Comment:
we usually register metrics in the `init` or alike. Will this even work? Since you'll register the same view for every scraper.

Reply:
I'll move this to `init`.